Develop

oscarnava opened the pull request:
Course: HTML & CSS3
Second microverse project

This is the first version of the second project for the Microverse training program.
It was developed by Nico and Oscar (the dream team) using basic html tags, form tags,
image map tags and CSS styling.

It's a functional copy of the login from the site mint.com. Not every function was
duplicated, but we tried to replicate the structure, the look&feel and validation of the
original site.

oscarnava committed:
Initial structure and content; no styles yet

nicolasmarino99 committed:
added styles to form

oscarnava committed:
Added format to inputs, labels and some other display elements

nicolasmarino99 committed:
added footer and button formating. Solved font awesome problem

oscarnava committed:
Preliminary version with funcional form, styling and documentation, ready for review

oscarnava committed:
Update README file

Corrected link for the original form

maelfosso commented:
Hello,

You did a really job, but before moving forward there are things to change

- [ ] Please, correct errors from https://validator.w3.org

- [ ] You should put the **Sign In** button, all the header is missing.

- [ ] The main Logo is one thing, the product is some thing else. You should separate them. For the product, you should have their image apart from the text.

- [ ] I encourage you to use `<main>` to express the main part of your code. It should contain the `<div class="header">` and `<div id="login-container">`

Kindly, make those changes and submit for a new code review request.

nicolasmarino99 committed:
solving reviwer tasks

oscarnava committed:
Added some responsive behaviour to the signin

oscarnava committed:
Solved issue with products text and icons

oscarnava commented:
As requested, we extracted the individual icons for the products and added the label in plain text format. We just want to make clear that the original site had the text merged with the icons, so that's why we did it that way. We also merged the branch to the develop branch, but we were unable to delete the "code-review-changes" branch because it still has open requests.

bolah2009 commented on README.md:
```diff
@@ -1 +1,10 @@
+
+It's a functional copy of the site [Mint.com](https://accounts.intuit.com/signup.html?offering_id=Intuit.ifs.mint&namespace_id=50000026&redirect_url=https%3A%2F%2Fmint.intuit.com%2Foverview.event%3Futm_medium%3Ddirect%26cta%3Dhero_sign_up_free_ProspectWeb%26adobe_mc%3DMCMID%253D08497889761544046421391270368961118431%257CMCAID%253D2E930D15050307F2-400011802000029C%257CMCORGID%253D969430F0543F253D0 ...
+repicated, but we tried to replicate the structure, the look&feel and validation of the
+original site.
```
`AWESOME`

Nice job writing a descriptive `README.md` file for describing what the project does. You can even add other information like live link and a screenshot of the webpage. :+1: :star2:

bolah2009 commented on index.html:
```diff
@@ -0,0 +1,84 @@
+        </div>
+        <form id="login-form" action="#" method="post">
+
+            <label for="email-ctl">
```
`AWESOME`

Good job associating the `<label>` with an `<input>` element. :clap: :+1: 

> #### Associating a `<label>` with an `<input>` element offers some major advantages:
> 
> - The label text is not only visually associated with its corresponding text input; it is programmatically associated with it too. This means that, for example, a screen reader will read out the label when the user is focused on the form input, making it easier for an assistive technology user to understand what data should be entered.
> 
> - You can click the associated label to focus/activate the input, as well as the input itself. This increased hit area provides an advantage to anyone trying to activate the input, including those using a touch-screen device.

Alternatively, you can nest the `<input>` directly inside the `<label>` like this: 

``` html
  <label for="email">Email address <br>
       <input type="email" name="email" id="email">
  </label> 
```


You can learn more [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label)

bolah2009 commented on index.html:
```diff
@@ -0,0 +1,84 @@
+            <a href="https://accounts.intuit.com/signup.html?offering_id=Intuit.ifs.mint&namespace_id=50000026&redirect_url=https%3A%2F%2Fmint.intuit.com%2Foverview.event%3Futm_medium%3Ddirect%26cta%3Dhero_sign_up_free_ProspectWeb%26adobe_mc%3DMCMID%253D08497889761544046421391270368961118431%257CMCORGID%253D969430F0543F253D0A4C98C6%252540AdobeOrg%257CTS%253D1562778150%26ivid%3D538f0975-5631-4a09- ...
+
+            <div id="link-imgs">
+                <a href="https://www.mint.com/"><img src="img/mint.png" alt="mint"><span class="product-name">mint</span></a>
```
`AWESOME`

Nice job, :clap: :star2:  adding an `alt` attribute to the `<img>` element, this is a very important accessibility attribute.
`Examples of scenarios where users benefit from text alternatives for images` are listed below and could also be found [here](https://www.w3.org/TR/html51/semantics-embedded-content.html#alt-text)

- They have a very slow connection and are browsing with images disabled.
- They have a vision impairment and use text to speech software.
- They have a cognitive impairment and use text to speech software.
- They are using a text-only browser.
- They are listening to the page being read out by a voice Web browser.
- They have images disabled to save on download costs.
- They have problems loading images or the source of an image is wrong.


##### SOURCE: [W3C Recommendation (Embedded Content)](https://www.w3.org/TR/html51/semantics-embedded-content.html#alt-text)

bolah2009 commented on index.html:
```diff
@@ -0,0 +1,84 @@
+        </div>
+    </div>
+</main>
+    <footer>
```
`AWESOME`

Very good job using `<footer>` element for the footer content, this helps to communicate meaning to the browser, developer, reader, and any other technologies interpreting the document (e.g. voice assistants). :ok_hand:

bolah2009 reviewed (approved):
# Congratulations [Dream Team](#)!  :tada:

You've done an awesome Job :muscle:

I'm impressed and proud of your progress, you should be too :clap:

### Important: I strongly recommend you make your sign in button align with the text like the picture below before you merge.

### Current

![clone intuit](https://user-images.githubusercontent.com/36057474/61458831-55ed9200-a963-11e9-9682-d2ac6c09f406.png)

### Preffered 

![original site](https://user-images.githubusercontent.com/36057474/61458850-600f9080-a963-11e9-87cb-7dbc5592f557.png)


Good luck with your next project
Keep it up :star2:

Cheers
Bola

oscarnava committed:
Fixed top right button to match the text height
